aws/credentials/stscreds/assume_role_provider.go

```diff
@@ -294,6 +294,7 @@ func (p *AssumeRoleProvider) Retrieve() (credentials.Value, error) {
 		AccessKeyID:     *roleOutput.Credentials.AccessKeyId,
 		SecretAccessKey: *roleOutput.Credentials.SecretAccessKey,
 		SessionToken:    *roleOutput.Credentials.SessionToken,
+		Expiration:      *roleOutput.Credentials.Expiration,
```

Comment:
There are several other credential providers in the SDK that could benefit from this feature.
* [ec2rolecreds](https://docs.aws.amazon.com/sdk-for-go/api/aws/credentials/ec2rolecreds/)
* [endpointcreds](https://docs.aws.amazon.com/sdk-for-go/api/aws/credentials/endpointcreds/)
* [plugincreds](https://docs.aws.amazon.com/sdk-for-go/api/aws/credentials/plugincreds/)
* [processcreds](https://docs.aws.amazon.com/sdk-for-go/api/aws/credentials/processcreds/)


One issue with adding `Expiration` to each of the providers, is ensuring all providers set this field correctly. This partly is due to a flaw in the V1 SDK's `credentials.Credentials` type. It abstracts away the concept of expiration from the credentials and providers. I'd like to find a way to improve this so the burden in not on the credential providers to add this behavior.  The credential providers already keep track of when credentials expire, but this is not exposed in the `credentials.Credentials` type. It may make sense to update `credentials.Credentials.Get` to populate the `Expiration` member of `credentials.Value`